pkg/destroy/aws/aws.go

```diff
@@ -140,16 +140,28 @@ func (o *ClusterUninstaller) Run() error {
 						&resourcegroupstaggingapi.GetResourcesInput{TagFilters: tagFilters},
 						func(results *resourcegroupstaggingapi.GetResourcesOutput, lastPage bool) bool {
 							for _, resource := range results.ResourceTagMappingList {
-								arn := *resource.ResourceARN
-								if _, ok := deleted[arn]; !ok {
+								arnString := *resource.ResourceARN
+								if _, ok := deleted[arnString]; !ok {
+									arnLogger := o.Logger.WithField("arn", arnString)
 									matched = true
-									err := deleteARN(awsSession, arn, filter, o.Logger)
+									parsed, err := arn.Parse(arnString)
```

Comment:
this change makes parsing ARN code exist in 2 different places
https://github.com/openshift/installer/pull/1704/files#diff-ab65f0e2ba0237c5cc1429673c22cc08R147
and
https://github.com/openshift/installer/pull/1704/files#diff-ab65f0e2ba0237c5cc1429673c22cc08R208
when we give that parseARN to deleteARN... IMO this is not ideal.

Reply:
> this change makes parsing ARN code exist in 2 different places

Yup.  I didn't think that was a big deal, it's not very complicated code.